cdap-cli/src/main/java/co/cask/cdap/cli/command/metadata/GetMetadataCommand.java

```diff
@@ -51,30 +50,49 @@ public GetMetadataCommand(CLIConfig cliConfig, MetadataClient client) {
 
   @Override
   public void perform(Arguments arguments, PrintStream output) throws Exception {
-    EntityId entity = EntityId.fromString(arguments.get(ArgumentName.ENTITY.toString()));
+    MetadataEntity metadataEntity =
+      MetadataCommandHelper.getMetadataEntity(arguments.get(ArgumentName.ENTITY.toString()));
     String scope = arguments.getOptional(ArgumentName.METADATA_SCOPE.toString());
-    Set<MetadataRecord> metadata = scope == null ? client.getMetadata(entity) :
-      client.getMetadata(entity, MetadataScope.valueOf(scope.toUpperCase()));
+    Table table;
+    try {
+      // for backward compatibility if the user is trying to retrieve the metadata for EntityIds return the output
+      // in old format i.e. MetadataRecord the EntityiId.fromMetadataEntity will fail with an IllegalArgumentException
+      // if the metadataEntity is not a cdap entity
+      EntityId entityId = EntityId.fromMetadataEntity(metadataEntity);
+      Set<MetadataRecord> metadata = scope == null ? client.getMetadata(entityId) :
+        client.getMetadata(entityId, MetadataScope.valueOf(scope.toUpperCase()));
 
-    Table table = Table.builder()
-      .setHeader("entity", "tags", "properties", "scope")
-      .setRows(
-        Iterables.transform(metadata, new Function<MetadataRecord, List<String>>() {
-          @Nullable
-          @Override
-          public List<String> apply(MetadataRecord record) {
-            return Lists.newArrayList(
-              record.toString(),
-              Joiner.on("\n").join(record.getTags()),
-              Joiner.on("\n").withKeyValueSeparator(":").join(record.getProperties()),
-              record.getScope().name());
-          }
-        })
-      ).build();
+      table = getTableBuilder()
+        .setRows(
+          metadata.stream().map(record -> Lists.newArrayList(
+            record.toString(),
+            Joiner.on("\n").join(record.getTags()),
+            Joiner.on("\n").withKeyValueSeparator(":").join(record.getProperties()),
+            record.getScope().name())).collect(Collectors.toList())
+        ).build();
+    } catch (IllegalArgumentException e) {
+      // the get metadata was called for some custom entity so it is okay to return in new format as we don't care
+      // about backward compatibility for calls for custom entity
+      Set<MetadataRecordV2> metadata = scope == null ? client.getMetadata(metadataEntity) :
+        client.getMetadata(metadataEntity, MetadataScope.valueOf(scope.toUpperCase()));
 
+      table = getTableBuilder()
+        .setRows(
+          metadata.stream().map(record -> Lists.newArrayList(
```

Comment:
The processing here looks to be the same as the processing in the above section. Can you refactor to eliminate duplicate code?